code/game/objects/objs.dm

```diff
@@ -79,22 +80,24 @@
 		AI = usr
 		if(AI.client && AI.interactee == src)
 			is_in_use = TRUE
-			attack_ai(AI)
+			if(interaction_flags & INTERACT_UI_INTERACT)
+				ui_interact(AI)
+			else
+				interact(AI)
 
-	for(var/mob/M in viewers(1, src))
+	for(var/mob/M in range(1, src))
```

Comment:
This chance means the proc will fail if someone is inside a locker, for example. Intended?